### PR TITLE
Fix mypy issues

### PR DESCRIPTION
**Pull request by rajaths010494**
https://app.circleci.com/pipelines/github/astronomer/astronomer-providers/4007/workflows/66592839-4f18-4ca7-8ea4-a64e46f8fd36/jobs/17639

image

### CODE REVIEW

File: astronomer/providers/cncf/kubernetes/operators/kubernetes_pod.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def execute(self, context: Context) -> None:  # noqa: D102
         self.pod: k8s.V1Pod = self.get_or_create_pod(self.pod_request_obj, context)
         self.defer()
 
-    def execute_complete(self, context: Context, event: Dict[str, Any]) -> Any:
+    def execute_complete(self, context: Context, event: Dict[str, Any]) -> Any:  # type: ignore[override]
```
Comment (pankajastro) [inline astronomer/providers/cncf/kubernetes/operators/kubernetes_pod.py:84]:
```suggestion
    def execute_complete(self, context: Context, event: Dict[str, Any]) -> Any
```

Comment (pankajastro) [reply astronomer/providers/cncf/kubernetes/operators/kubernetes_pod.py:84]: can you please try above ^^

File: astronomer/providers/google/cloud/operators/dataproc.py [hunk 2]
```diff
@@ -379,7 +379,7 @@ def execute(self, context: "Context") -> None:
             method_name="execute_complete",
         )
 
-    def execute_complete(self, context: Dict[str, Any], event: Optional[Dict[str, Any]] = None) -> Any:
+    def execute_complete(self, context: Dict[str, Any], event: Optional[Dict[str, Any]] = None) -> Any:  # type: ignore[override]
```
Comment (pankajastro) [inline astronomer/providers/google/cloud/operators/dataproc.py:382]:
```suggestion
    def execute_complete(self, context: Context, event: Dict[str, Any]) -> Any
```

Comment (pankajastro) [reply astronomer/providers/google/cloud/operators/dataproc.py:382]: can you please try above ^^

**Comment by pankajastro** [comment]
We should match the signature of `execute_complete` with the parent class `execute_complete ` method instead of ignoring mypy, WDYT?

**Comment by rajaths010494** [comment]
> We should match the signature of `execute_complete` with the parent class `execute_complete ` method instead of ignoring mypy, WDYT?

Agreed where ever we can we should match to parent class. But in OSS they are certain classes use 
`def execute_complete(self, context: Context, event: dict, **kwargs):`  if we use this signature mypy will still throw error because dict should be defined as `Dict[str, Any]`. Let me check if the changes can be made same as parent class.

**Review by pankajkoti — APPROVED** [review]

**Review by pankajastro — APPROVED** [review]

**Pull request merged**